src/foam/u2/view/ChoiceView.js

```diff
@@ -259,7 +259,7 @@ foam.CLASS({
 
     function fromProperty(p) {
       this.SUPER(p);
-      this.defaultValue = p.value;
+      this.defaultValue = p.value && this.findChoiceByData(p.value);
```

Comment:
`defaultValue` doesn't appear to be used anymore after your changes and can be removed.

Reply:
@TW80000 without defaultvalue, the data won't be displayed in the detail view..